src/config/spark.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+
+# Copyright 2024 Canonical Limited
+# See LICENSE file for licensing details.
+
+"""Spark related configurations."""
+
+from typing import Optional
+
+from lightkube import Client
+
+from constants import KYUUBI_OCI_IMAGE
+from s3 import S3ConnectionInfo
+from utils.logging import WithLogging
+
+
+class SparkConfig(WithLogging):
```

Comment:
Last time, we realized that Kyuubi would not read the properties tagged with a service account by default (since Kyuubi uses its own Spark binaries and not the binaries that wrap the Spark8t library).

Later we discussed about the possibility of Kyuubi reading the service account props directly from secrets, and then the charm preparing the file on-the-fly. 

I didn't include that as part of this ticket because I believe that constitutes the tasks we do during integration of Kyuubi with Integration Hub. I didn't want to extend the scope of this ticket at the moment and would like to pick this as part of Kyuubi <> Integration Hub integration task that we'll pick next week. Do you agree?


P.S. Even when we add logic to read the Spark props from Kubernetes secret, I think maybe the Kyuubi configuration and Hive configurations still need to be constructed by charm itself?

Reply:
As you prefer, I'm also happy to address it in the next ticket, but I want this to land. I feel we are missing to centrilize code and business logic otherwise. 

> Later we discussed about the possibility of Kyuubi reading the service account props directly from secrets, and then the charm preparing the file on-the-fly.

Yes, that's what I meant. We won't put any spark-client wrapper around kyuubi, but we can use spark8t functionalities for creating the spark props file. 

> I think maybe the Kyuubi configuration and Hive configurations still need to be constructed by charm itself?

Yes, totally! These are Kyuubi specific right now

Reply:
I agree with you that this is the way we want to go forward with. It is just a matter of whether we do it during dedicated task for Kyuubi <> Integration Hub next week, or extend this and move this ticket to the next pulse. I prefer to address these during integration ticket we pick next, but I'm okay to address these here if you prefer it otherwise.

Reply:
I feel that it is a bit more general than just integration with Integration Hub. I mean, this is needed also when we integrate with a service account that was configured using spark-client. Hence the comment. 

I'm not sure is fully in scope for the ticket of integration hub, as the task is more an integration tests to be implemented in spark-k8s-bundle. When we integrate, we may have to fix UX elsewhere, but maybe this feature (meaning that kyuubi should be able to read configuration that a service account comes with) could already be implemented. 

Maybe the right call would be to add a ticket for this specifically if you like, that we can add to the pulse. It should not be a huge ticket, actually mostly replacing this class with the call above

Reply:
I have created [this](https://warthogs.atlassian.net/browse/DPE-4165) ticket. I believe it is fair to address this in a proper ticket. I have also provided a bit of description. Let's follow up on MM or on the ticket if there is unclarity. Resolving the comment now